src/crypto/CHIPCryptoPAL.h

```diff
@@ -262,6 +262,16 @@ CHIP_ERROR add_entropy_source(entropy_source fn_source, void * p_source, size_t
  **/
 CHIP_ERROR pbkdf2_sha256(const uint8_t * password, size_t plen, const uint8_t * salt, size_t slen, unsigned int iteration_count,
                          uint32_t key_length, uint8_t * output);
+
+/** @brief Generate a new keypair for ECP256 curve.
+ * @param pubkey Generated public key
+ * @param pklen length of pubkey buffer
```

Comment:
will you document the in/out behavior for these len fields?

Reply:
e.g. can you query the size required?  it looks like you have to guess and it's an error to get it wrong

Reply:
fwiw these outputs are static sizes.

Reply:
No way to query the size from underlying crypto library. The API does allow the caller to provide over allocated buffer. The size is updated based on how much is required.